R/magpie-class.R

```diff
@@ -302,39 +305,27 @@ setMethod("[", #nolint
       j <- .dimextract(x, j, 2, pmatch = pmatch, invert = invert)
     }
     if (!missing(k)) k <- .dimextract(x, k, 3, pmatch = pmatch, invert = invert)
-
-    if (!missing(i) && missing(j) && !missing(k) && .isFALSE(k) &&
-      .isFALSE(drop) && .isFALSE(pmatch) && .isFALSE(invert)) {
-      # there is a weird case in which k is actually missing but is getting the
-      # value of the next argument in line (drop)
-      # this one is catched via .isFALSE(k)
-      # in addition non-default settings for drop, pmatch and invert indicate that object should still be
-      # returned as magpie object (and not as numeric as the following line will do)
+    if (length(sys.call(-1)) == 3) {
```

Comment:
Please add a comment what is happening here and why

Reply:
done